pkg/installer/etchost/etchost_assets.go

```diff
@@ -0,0 +1,34 @@
+package etchost
+
+import (
+	"github.com/openshift/installer-aro-wrapper/pkg/installer/dnsmasq"
+	"github.com/openshift/installer/pkg/asset/ignition/bootstrap"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+)
+
+func AppendEtcHostFiles(bootstrapAsset *bootstrap.Bootstrap, installConfig installconfig.InstallConfig, dnsConfig dnsmasq.DNSConfig) error {
+	etcHostIgnConfig, err := EtcHostsIgnitionConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP)
+	if err != nil {
+		return err
+	}
+	bootstrapAsset.Config.Storage.Files = append(bootstrapAsset.Config.Storage.Files, etcHostIgnConfig.Storage.Files...)
+	bootstrapAsset.Config.Systemd.Units = append(bootstrapAsset.Config.Systemd.Units, etcHostIgnConfig.Systemd.Units...)
+	etcHostMasterMachineConfig, err := EtcHostsMachineConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, "master")
+	if err != nil {
+		return err
+	}
+	etcHostWorkerMachineConfig, err := EtcHostsMachineConfig(installConfig.Config.ClusterDomain(), dnsConfig.APIIntIP, dnsConfig.GatewayDomains, dnsConfig.GatewayPrivateEndpointIP, "worker")
+	if err != nil {
+		return err
+	}
+
+	err = dnsmasq.AppendMachineConfigToBootstrap(etcHostMasterMachineConfig, bootstrapAsset, "/opt/openshift/openshift/99_openshift-machineconfig_99-master-aro-etc-hosts-gateway-domains.yaml")
```

Comment:
Reuses functions from the dnsmasq PR 
https://github.com/openshift/installer-aro-wrapper/blob/d6af50321aced4a322a645539afa39420b679469/pkg/installer/dnsmasq/dnsmasq_config.go#L40